### PR TITLE
chore: refactor handling logic for embeds in app router and add a test

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Refactors `withEmbedSsrAppDir` and reduces code complexity 
- Adds a unit test

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Please use the latest Vercel preview and test please 🙏.

### CODE REVIEW

File: apps/web/app/future/org/[orgSlug]/embed/page.tsx [hunk 1]
```diff
@@ -1,28 +1,14 @@
...
 
-import { buildLegacyCtx } from "@lib/buildLegacyCtx";
 import { getServerSideProps } from "@lib/team/[slug]/getServerSideProps";
 
 import type { PageProps } from "~/team/team-view";
 import TeamPage from "~/team/team-view";
 
-const getData = withAppDirSsr<PageProps>(getServerSideProps);
-
-export const generateMetadata = async ({ params, searchParams }: _PageProps) => {
```
Comment (hbjORbj) [inline apps/web/app/future/org/[orgSlug]/embed/page.tsx:16]: generateMetadata is not needed in embeds

File: apps/web/app/future/[user]/embed/page.tsx [hunk 2]
```diff
@@ -7,10 +6,6 @@ import { getServerSideProps } from "@server/lib/[user]/getServerSideProps";
 import type { PageProps as UserPageProps } from "~/users/views/users-public-view";
 import LegacyPage from "~/users/views/users-public-view";
 
-export { generateMetadata } from "../page";
```
Comment (hbjORbj) [inline apps/web/app/future/[user]/embed/page.tsx:10]: generateMetadata is not needed in embeds

File: apps/web/app/future/[user]/[type]/embed/page.tsx [hunk 3]
```diff
@@ -1,15 +1,10 @@
-import { withAppDirSsr } from "app/WithAppDirSsr";
 import withEmbedSsrAppDir from "app/WithEmbedSSR";
 import { WithLayout } from "app/layoutHOC";
 
 import { getServerSideProps } from "@server/lib/[user]/[type]/getServerSideProps";
 
 import LegacyPage, { type PageProps } from "~/users/views/users-type-public-view";
 
-export { generateMetadata } from "../page";
```
Comment (hbjORbj) [inline apps/web/app/future/[user]/[type]/embed/page.tsx:9]: generateMetadata is not needed in embeds

File: apps/web/app/WithEmbedSSR.tsx [hunk 4]
```diff
@@ -9,48 +7,46 @@ export type EmbedProps = {
...
-        // Make sure that redirect happens to /embed page and pass on embed query param as is for preserving Cal JS API namespace
-        const newDestinationUrl = `${urlPrefix}${destinationUrlObj.pathname}/embed?${
-          destinationQueryStr ? `${destinationQueryStr}&` : ""
-        }layout=${layout}&embed=${embed}`;
-
-        redirect(newDestinationUrl);
-      }
+    const ssrResponse = await getServerSideProps(context);
+
+    if ("redirect" in ssrResponse) {
```
Comment (hbjORbj) [inline apps/web/app/WithEmbedSSR.tsx:16]: Copied the logic from withEmbedSsr in `lib/withEmbedSsr.ts` (which is currently used for pages router routes) and adapted it to App Router

**Comment by hbjORbj** [comment]
@hbjORbj Add a unit test suite for `withEmbedSsrAppDir`

**Review by eunjae-lee — APPROVED** [review]

**Pull request merged**